### DISCUSSION TITLE
Copy Placeholder for Devanagari

### DISCUSSION THREAD

**Discussion by AmitPatil999** [viresh-ratnakar/exolve#99] [Ideas]
Hi Folks, This is a very good software you have developed and I am trying to use it for publishing Marathi crosswords.  I tried the copy placeholder feature but didn't work. (I have specified options as hi Devanagari.) It gives me message as "Are you sure you want to do this mismatched copy?"
This is very important feature for Indian languages as Indian words are best written as whole words using transliteration, than individual letters.
Thanks

**Comment by viresh-ratnakar** [comment]
Can you please send me a link to crossword that you're trying to get to work? Because I'm not sure I understand the issue. Exolve does not support transliteration. You'd want to use something like Google Input Tools to convert transliterated text into Devanagari.

I'm assuming you've looked at the documentation in https://github.com/viresh-ratnakar/exolve/blob/master/README.md#exolve-language. There's also a test Hindi crossword that I have created: https://viresh-ratnakar.github.io/dev/test-hindi.html

Also, I think you should be able to use the Marathi language code (mr Devanagari).

I would be very happy to help, once I understand the issue fully...

**Reply by AmitPatil999** [reply]
Hi Viresh,
Thanks for the quick reply.
Here my app = https://marathi-crossword-9b436.web.app/
So the issue here is- although possible, it's not easy to type Devanagari composite letters into cells. One has to double click the cell and then they type composite letters. Also, it's not easy to type composite letters. 
But it's easy to type the whole word into placeholder using Google Input tools' transliterate. But Exolve doesn't support copying Devanagari from placeholder into cells. It throws length mismatch error. So my question is - is it possible for exolve to support copying devanagari composite letters from placeholder to cells. 
Also, is there a way user can share their result after playing, in the form of colored boxes, similar to Wordle?

**Reply by viresh-ratnakar** [reply]
Hmm.. I thought I had replied here, but evidently not :-).

I did push a change that should allow easier copying from placeholder blanks. Please note this documentation that I added to README.md:

You may also want to provide [placeholder blanks](https://github.com/viresh-ratnakar/exolve/blob/master/README.md#forcing-the-display-of-placeholder-blanks) for languages such as Hindi that have compound letters. Exolve recognizes multi-character letters separated by spaces within placeholders. So, if you have a 3-letter clue, and the placeholder text contains "उ स्ता द" then copying this into a 3-letter entry will place "उ" in the first cell, "स्ता" in the second cell, and "द" in the third cell.

**Comment by AmitPatil999** [comment]
Hi Viresh, Thanks for responding. This new change sounds awesome. I will have a look at it. If you don't mind, I'd like to suggest to remove the "space between letters" criteria, if possible. It requires users to click between the letters and hit space which can be a big turn off for most users. But That's just my opinion.
Also, I created my own front-end. Here - [https://marathi-word-games.web.app/](https://marathi-word-games.web.app/)
Thanks,

**Comment by viresh-ratnakar** [comment]
Thanks, and you site looks great!

The reason I need the spaces is that without them I cannot easily split the string into character groups. I'll need to add more complexity to be able to parse (e.g.) "उस्ताद" as "उ स्ता द" from the constituent unicode characters, which are:

"उ"
"स"
"्"
"त"
"ा"
"द"

**Comment by AmitPatil999** [comment]
Hi Viresh,
Thank for the kind words. Oh! yes, one needs the Devanagari splitting logic. I have done that. Let me know if you want the code. Reach out to me at vegapoint @ gmail . com

By the way, I am looking for database for Marathi words and clues for my crossword. Any idea, where I can get it?
Thanks,

**Comment by AmitPatil999** [comment]
Hi Viresh, 
Do you know of any algorithm that can build crossword given the database of words. 
Thanks,

**Comment by viresh-ratnakar** [comment]
Well, my other crossword project, Exet
<https://github.com/viresh-ratnakar/exet>, can certainly do it. I've tested
it a fair bit for English, for which I have a good/comprehensive dictionary:

Here's my web app location for it:
https://viresh-ratnakar.github.io/exet.html

I also have it working for Hindi, but I only could find a small open source
dictionary for Hindi, so auto-fill doesn't work that well:
https://viresh-ratnakar.github.io/exet-hindi.html

If you can find a good/comprehensive open-source dictionary, then it should
work pretty well. The dictionary will need a bit of preprocessing (see my
Lufz <https://github.com/viresh-ratnakar/lufz> project) but I can help with
that.

Viresh

**Comment by AmitPatil999** [comment]
Thanks Viresh,
I checked Exet. But as I understand, it still needs human intervention to create crossword. What I have done is - on autopilot. It doesn't require any human intervention. It takes words from database in spreadhsheet and build crossword using the static layout.

**Comment by viresh-ratnakar** [comment]
Sure, Exet has that fully automatic option too (Edit > Autofill) that works
well in English. That's what I meant—but the only open source lexicon for
Hindi that I found is limited, so I'm not sure how well it (Autofill) works
for Hindi.
